### PR TITLE
Console: Continue building the environment flows UI

### PR DESCRIPTION
**Pull request by ThibaudAV**
## Issue

https://gravitee.atlassian.net/browse/APIM-5388

## Description

- Init & impl add-edit dialog
image

- init env-flow studio page 
image

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-zrdzisljjo.chromatic.com)

### CODE REVIEW

File: gravitee-apim-console-webui/src/management/settings/environment-flows/environment-flows.component.html [hunk 1]
```diff
@@ -67,7 +67,7 @@
           <th mat-header-cell *matHeaderCellDef></th>
           <td mat-cell *matCellDef="let element">
             <button
-              *gioPermission="{ anyOf: ['environment-flows-d'] }"
+              *gioPermission="{ anyOf: ['environment-environment_flows-d'] }"
               type="button"
               mat-button
               aria-label="Button to remove"
```
Comment (ytvnr) [inline gravitee-apim-console-webui/src/management/settings/environment-flows/environment-flows.component.html]: "Remove the environment flow" ?

File: gravitee-apim-console-webui/src/management/settings/environment-flows/environment-flows-add-edit-dialog/environment-flows-add-edit-dialog.component.ts [hunk 2]
```diff
@@ -16,22 +16,81 @@
...
-export type EnvironmentFlowsAddEditDialogResult = boolean;
+export interface EnvironmentFlowsAddEditDialogData {
+  apiType: ApiV4['type'];
+}
+
+export type EnvironmentFlowsAddEditDialogResult = undefined | { name: string; description?: string; phase: ExecutionPhase };
+
+const PHASE_BY_API_TYPE: Record<ApiV4['type'], ExecutionPhase[]> = {
+  PROXY: ['REQUEST', 'RESPONSE'],
+  MESSAGE: ['REQUEST', 'MESSAGE_REQUEST', 'MESSAGE_RESPONSE', 'RESPONSE'],
```
Comment (ytvnr) [inline gravitee-apim-console-webui/src/management/settings/environment-flows/environment-flows-add-edit-dialog/environment-flows-add-edit-dialog.component.ts]: Maybe in this order ?
['REQUEST', 'RESPONSE', 'MESSAGE_REQUEST', 'MESSAGE_RESPONSE']
Or this one ?
['REQUEST', 'MESSAGE_REQUEST', 'RESPONSE', 'MESSAGE_RESPONSE']

File: gravitee-apim-console-webui/src/management/settings/environment-flows/environment-flows-add-edit-dialog/environment-flows-add-edit-dialog.component.ts [hunk 3]
```diff
@@ -16,22 +16,81 @@
...
+    this.isValid$ = this.formGroup.statusChanges.pipe(
+      startWith(this.formGroup.status),
+      map((status) => status === 'VALID'),
+    );
+  }
+
+  protected onSave(): void {
+    if (this.formGroup.invalid) {
+      return;
+    }
```
Comment (ytvnr) [inline gravitee-apim-console-webui/src/management/settings/environment-flows/environment-flows-add-edit-dialog/environment-flows-add-edit-dialog.component.ts:89]: is this case possible ? if we call onSave it mean the button is enabled, no ?

File: gravitee-apim-console-webui/src/management/settings/environment-flows/environment-flows.component.html [hunk 4]
```diff
@@ -20,8 +20,19 @@
   <mat-card-header>
     <mat-card-title>Environment Flows</mat-card-title>
     <mat-card-subtitle>Flows allow to apply different policies on your API event phases</mat-card-subtitle>
-      <button mat-raised-button color="primary" (click)="onAddEnvironmentFlow()"><mat-icon svgIcon="gio:plus"></mat-icon>Add Flow</button>
     <div class="environment-flows__headerRightBtn" *gioPermission="{ anyOf: ['environment-environment_flows-c'] }">
+      <button mat-raised-button color="primary" [matMenuTriggerFor]="addEnvironmentFlowMenu">
+        <mat-icon svgIcon="gio:plus"></mat-icon>Add Flow
```
Comment (ytvnr) [inline gravitee-apim-console-webui/src/management/settings/environment-flows/environment-flows.component.html]: Add Environment Flow

File: gravitee-apim-console-webui/src/management/settings/environment-flows/environment-flows-studio/environment-flows-studio.component.html [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+
+<div class="back-button">
+  <a mat-button [routerLink]="'../../'"><mat-icon svgIcon="gio:arrow-left"></mat-icon>Go back to your endpoints</a>
```
Comment (ytvnr) [inline gravitee-apim-console-webui/src/management/settings/environment-flows/environment-flows-studio/environment-flows-studio.component.html]: Go back to the settings ?

**Review by ytvnr — APPROVED** [review]

File: gravitee-apim-console-webui/src/management/settings/environment-flows/environment-flows-add-edit-dialog/environment-flows-add-edit-dialog.component.ts (hunk 2 above)
Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/management/settings/environment-flows/environment-flows-add-edit-dialog/environment-flows-add-edit-dialog.component.ts]: I originally had the 1st order . but i changed it for some reason haha. there's no case where the response executes at the end of the message exchange. but that must depend on your endpoint. right ?

The both suits me. as you prefer ?

File: gravitee-apim-console-webui/src/management/settings/environment-flows/environment-flows-add-edit-dialog/environment-flows-add-edit-dialog.component.ts (hunk 3 above)
Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/management/settings/environment-flows/environment-flows-add-edit-dialog/environment-flows-add-edit-dialog.component.ts:89]: it allow to focuses the user on the 1st invalid form field. if i disable the button it doesn't work. the button change color when it's valid or not

File: gravitee-apim-console-webui/src/management/settings/environment-flows/environment-flows-studio/environment-flows-studio.component.html (hunk 5 above)
Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/management/settings/environment-flows/environment-flows-studio/environment-flows-studio.component.html]: oups copy past

**Pull request merged**